### PR TITLE
Add S3 resources for GitLab.

### PR DESCRIPTION
**Pull request by justin-o12**

### CODE REVIEW

**Comment by justin-o12** [comment]
Once the [VPC PR](https://github.com/defenseunicorns/uds-infrastructure-software-factory-aws/pull/2) is merged, I will incorporate standard testing and CI in this PR.

**Review by zachariahmiller — APPROVED** [review]
lgtm. we'll also need the outputs though to provide into zarf. That can be next steps though i think.

File: gitlab/example.tfvars [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bucket_name_suffix = "-example"
```
Comment (zachariahmiller) [inline gitlab/example.tfvars:1]: Should the dash be expected to be passed in or in the module call like `  bucket        = "${var.bucket_name_prefix}-${each.key}-${var.bucket_name_suffix}"`?

**Review by zachariahmiller — COMMENTED** [review]
I think we need the bucket names as outputs dont we?

File: gitlab/example.tfvars (hunk 1 above)
Comment (justin-o12) [reply gitlab/example.tfvars:1]: I suggest passing in the dash for suffix and prefix to keep them optional.  If the variables are optional but we hardcode the dash, we might end up with a resource named `prefix-gitlabbucket-`.

Comment (zachariahmiller) [reply gitlab/example.tfvars:1]: yeah i understand. My part of brain hates that, but it does make sense without adding a bunch more logic. 😄 🚀

**Pull request merged**